apps/saleor-app-checkout/config/apl.ts

```diff
@@ -1,62 +1,60 @@
-import { FileAPL, UpstashAPL } from "@saleor/app-sdk/APL";
+import { FileAPL, UpstashAPL, SaleorCloudAPL, AuthData, APL } from "@saleor/app-sdk/APL";
 import invariant from "ts-invariant";
 import Fs from "fs/promises";
 import { unpackPromise } from "../utils/unpackErrors";
-import { CheckoutVercelAPL } from "./checkoutVercelApl";
 
-type Result = {
-  saleorApiUrl: string;
-  appToken: string;
-};
+export let apl: APL;
 
-const getApl = () => {
-  switch (process.env.APL) {
-    case "upstash":
-      return (() => {
-        const UPSTASH_REDIS_REST_URL = process.env.UPSTASH_REDIS_REST_URL;
-        const UPSTASH_REDIS_REST_TOKEN = process.env.UPSTASH_REDIS_REST_TOKEN;
+switch (process.env.APL) {
+  case "upstash":
+    const UPSTASH_REDIS_REST_URL = process.env.UPSTASH_REDIS_REST_URL;
+    const UPSTASH_REDIS_REST_TOKEN = process.env.UPSTASH_REDIS_REST_TOKEN;
 
-        invariant(UPSTASH_REDIS_REST_URL, "Missing UPSTASH_REDIS_REST_URL!");
-        invariant(UPSTASH_REDIS_REST_TOKEN, "Missing UPSTASH_REDIS_REST_TOKEN!");
+    invariant(UPSTASH_REDIS_REST_URL, "Missing UPSTASH_REDIS_REST_URL!");
+    invariant(UPSTASH_REDIS_REST_TOKEN, "Missing UPSTASH_REDIS_REST_TOKEN!");
 
-        return new UpstashAPL({
-          restURL: UPSTASH_REDIS_REST_URL,
-          restToken: UPSTASH_REDIS_REST_TOKEN,
-        });
-      })();
-    case "file":
-      void printFileAplWarning();
-      return new FileAPL();
-    case "vercel":
-      return new CheckoutVercelAPL();
```

Comment:
We're currently using this on staging and demo. What should we do now?

Reply:
Demo instance should be using SaleorCloud APL.
For custom deployments Upstash APL can be used.

In the future, a workflow will cover reading tokens from the environment. Before we'll be able to implement it, more changes in the core are necessary.

Reply:
There is a way to do it, but not "dynamically" with VercelAPL. This approach is a rabbit hole and we dropped it some time ago
We will work on the documentation. Officially we encourage to use other APL.

To achieve it for single tenant way (which we don't recommend):
1. console.log authdata in registerHandler
2. save it to env variables
3. write simple read-only "EnvApl" and read it from there

We don't recommend it because this flow will break with upcoming features like token rotation or upgrades

Hence even single tenant flow should use some kind of mutable storage

Reply:
But that's exactly what `CheckoutVercelAPL` did – read variables from env. So why is it removed?

Reply:
I just checked the code, indeed, it may be a mistake on our side
@krzysztofwolski can you check it again

Broader context - we removed build-in `VercelAPL` from SDK, which was actually complex and fragile logic, that was using external service (appraptor) hosted by us, that requires Vercel tokens to talk to Vercel API and update env variables on the fly. It was hacky and often broke.

In the future we want to enable Web API + VercelEdgeConfig to re-enable single tenant flow with Vercel

Reply:
Just talked to @krzysztofwolski  - we need to find a middle spot. Approach you did have in CheckoutVercelAPL is fine, but it needs more fields - Krzyh will update it

Reply:
@mmiszy Checkout Vercel APL is back. I've added fields required by the new APL interface.